src/_micro-api/libraries/signalDecoder/src/signalDecoder.cpp

```diff
@@ -43,6 +43,49 @@
 #endif
 
 
+///////////////////////////////////////////////////////////////////////
+/* SELFMADE -  Implementation of itoa() */
+
+char* myitoa(int num, char* str, int base) {
+  int i = 0;
+  bool isNegative = false;
+  /* Handle 0 explicitely, otherwise empty string is printed for 0 */
+  if (num == 0) {
+    str[i++] = '0';
+    str[i] = '\0';
+    return str;
+  }
+  // In standard itoa(), negative numbers are handled only with
+  // base 10. Otherwise numbers are considered unsigned.
+  if (num < 0 && base == 10) {
+    isNegative = true;
+    num = -num;
+  }
+  // Process individual digits
+  while (num != 0) {
+    int rem = num % base;
+    str[i++] = (rem > 9) ? (rem - 10) + 'a' : rem + '0';
+    num = num / base;
+  }
+
+  if (isNegative) str[i++] = '-';  // If number is negative, append '-'
+  str[i] = '\0';                   // Append string terminator
+  myreverse(str, i);               // Reverse the string
+  return str;
+}
+
+/* A utility function to reverse a string  */
+void myreverse(char str[], int length) {
```

Comment:
Ich würde hier auf char pointer wechseln. Was ist die maximale Länge die hier unterstüzzt werden soll?
Da es keine negative länge gibt würde ich in jedem Falle  auf unsigned wechseln.

Reply:
> Ich würde hier auf char pointer wechseln. Was ist die maximale Länge die hier unterstüzzt werden soll?

Dann mäkelt der Compiler: "no return statement in function returning non-void [-Werror=return-type]"
Wir geben ja auch keine Variable zurück, sondern manipulieren direkt str[]

Die maximale Länge dürfte bei int gleich 7 sein. Oder haben wir größere Zahlen zu verarbeiten?

> Da es keine negative länge gibt würde ich in jedem Falle auf unsigned wechseln.

Das habe ich probiert, bringt offensichtlich keinen Vorteil (vorher/nachher):

```
Der Sketch verwendet 26774 Bytes (93%) des Programmspeicherplatzes. Das Maximum sind 28672 Bytes.
Globale Variablen verwenden 874 Bytes des dynamischen Speichers.
ohne "bool isNegative" geht nicht so ohne weiteres
int in uint8_t - bringt nichts
Der Sketch verwendet 26776 Bytes (93%) des Programmspeicherplatzes. Das Maximum sind 28672 Bytes.
Globale Variablen verwenden 874 Bytes des dynamischen Speichers.
```

Reply:
wenn die maximale Länge 7 ist die verarbeitet werden soll, dann reicht auch ein uint8_t Datentyp aus.
Der Unterschied zwischen singed und unsigend macht sich nur im Programm selbst bemerkbar, nicht im Speicherverbrauch. ein int und ein unsigned int bestehen weiterhin aus 16 bit.